app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -420,10 +420,19 @@ def save_tags_generated_from_labels(entity, hashes, _target = nil)
                      :name          => label_hash[:name],
                      :value         => label_hash[:value])
     end
-    current_tags = labels.collect_concat { |label| ContainerLabelTagMapping.tags_for_label(label) }
-    mappable_tags = ContainerLabelTagMapping.mappable_tags
 
-    entity.tags = entity.tags - mappable_tags + current_tags
+    begin
+      current_tags = labels.collect_concat { |label| ContainerLabelTagMapping.tags_for_label(label) }
+      mappable_tags = ContainerLabelTagMapping.mappable_tags
+      entity.tags = entity.tags - mappable_tags + current_tags
+    rescue => err
+      raise if EmsRefresh.debug_failures
+      _log.error("Auto-tagging #{entity.class} [#{entity.name}] from labels [#{labels}] failed with error [#{err}].")
+      if err.is_a? ActiveRecord::RecordInvalid
+        _log.error("Validation error details: #{err.record.errors.details}") rescue nil
+      end
```

Comment:
I fear the chance of the three lines above raising another exception and us not having a stack trace... I would prefer removing them (we would be able to understand we have a validation error from the trace)